scripts/hooks/run-pre-commit.sh

```
@@ -0,0 +1,87 @@
+          echo "[pre-commit] Installing tools and hooks…" >&2
+          bash ./scripts/setup-dev.sh || true
+          if command -v dprint >/dev/null 2>&1; then
+            echo "[pre-commit] dprint installed; formatting JSON/YAML…" >&2
+            dprint fmt
+          else
+            echo "[pre-commit][WARN] dprint still not available; continuing." >&2
+          fi
+          ;;
+        *)
+          echo "OK… but you should REALLY consider installing them." >&2
+          echo "You can do so with:  make setup-dev\n\nContinuing… but grumbling." >&2
```

Shellcheck (0.11.0)

[info] echo may not expand escape sequences. Use printf.

(SC2028)